src/plugins/dataflow/components/dataflow-program.tsx

```diff
@@ -513,12 +520,24 @@ export class DataflowProgram extends BaseComponent<IProps, IState> {
     this.programEditor.clear();
   };
 
+  private createCaseIdForTick = (collectedTime: number) => {
+    return collectedTime + "*" + this.props.tileId;
+  };
+
   private tick = () => {
-    // Update the sampling rate
     const now = Date.now();
     this.setState({lastIntervalDuration: now - this.lastIntervalTime});
     this.lastIntervalTime = now;
 
+    // set up to record each node, should recording be on
+    const isRecording = this.props.programRecordState === 1;
+    const caseId = isRecording ? this.createCaseIdForTick(now) : null;
+    const existingAttributes = this.props.tileModel.existingAttributes();
+    if (isRecording){
+      this.props.tileModel.addNewCaseFromAttrKeys(existingAttributes, caseId as string);
+    }
```

Comment:
```suggestion
```
Creating the case here and setting its values below is inefficient.

Reply:
Got it, in this new push I have all the values being set within this block.